### DISCUSSION TITLE
Can I get the fp from a concatenated dataframe ?

### DISCUSSION THREAD

**Discussion by Lili-Cao** [chemosim-lab/ProLIF#198] [Q&A]
Hi,
I am trying to generate a barcode plot for interactions for a set of complexes.
I made a concatenated dataframe, which is very similar to the one in the tutorial for docking poses section. However, since I am not analysing all the  poses, I didn't have a fp object contain all fps. Therefore, I am not able to use the fp.plot_barcode()
function. 

Instead I get this error  "Please run the fingerprint analysis before attempting to display  results." which I could understood.

I am wondering, is there a way to get fps from a concatenated dataframe? So I can  make the barcode plot?

Thanks!
Lili

**Comment by cbouy** [comment]
Hi,

You can directly use the `Barcode` class for this (used by `fp.plot_barcode` under the hood):
```python
from prolif.plotting.barcode import Barcode

barcode = Barcode(concatenated_df)
barcode.display()
```

**Comment by Lili-Cao** [comment]
Thank you!
